### PR TITLE
Experimental testing

### PR DESCRIPTION
**Pull request by zcmoore**
Related scrum tasks: Abstract WidgetUnitTest, Widget Unit Test Framework, Integrate GLUnitTesting system with JUnit

Some of the resulting code from the above tasks was abandoned do to high complexity, low usability, and better alternative solutions. The alternative solutions are included in this merge, as well as prosperous results of the above tasks. The abandoned code can be viewed in the commit history, and will be later pushed to the garbage branch.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -61,3 +62,5 @@ build/
 
 ## OS Specific
 .DS_Store
+
+desktop/.classpath
```
Comment (jreigel) [inline .gitignore:66]: Warning for no newline at the end of file. Unsure of the ramifications from this.

File: desktop/build.gradle [hunk 2]
```diff
@@ -43,4 +57,4 @@ task afterEclipseImport(description: "Post processing after project generation",
     printer.setPreserveWhitespace(true)
     printer.print(classpath)
   }
-}
```
Comment (jreigel) [inline desktop/build.gradle:46]: Unsure again about the missing newline warning.

**Comment by jreigel** [comment]
Couldn't find any errors in the code, javadocs appeared to be correct. Unsure about the consequences from the missing newline warning messages on  the .gitignore and build.gradle.

File: desktop/build.gradle (hunk 2 above)
Comment (zcmoore) [inline desktop/build.gradle:46]: I'll run the formatter again

**Pull request merged**